packages/datadog-plugin-prisma/test/integration-test/client.spec.js

```diff
@@ -14,75 +14,152 @@ const {
+        return
+      }
 
-    before(function () {
-      variants = varySandbox('server.mjs', '@prisma/client', 'PrismaClient')
-    })
+      const supportedRange = config.configFile ? '>=7.0.0' : '<7.0.0'
 
-    afterEach(async () => {
-      proc?.kill()
-      await agent.stop()
-    })
+      withVersions('prisma', '@prisma/client', supportedRange, version => {
+        if (config.ts && version === '6.1.0') return
+        let variants
+        const paths = ['./packages/datadog-plugin-prisma/test/integration-test/*', config.schema]
+
+        if (isPrismaV7) paths.push(config.configFile)
+
+        const deps = [`prisma@${version}`, `@prisma/client@${version}`, 'typescript']
+        if (isPrismaV7) deps.push('@prisma/adapter-pg')
+        useSandbox(deps, false, paths)
+
+        before(function () {
+          variants = varySandbox(config.serverFile, config.importPath, 'PrismaClient')
+        })
 
-    for (const variant of varySandbox.VARIANTS) {
-      it('is instrumented', async function () {
-        this.timeout(60000)
-        const res = agent.assertMessageReceived(({ headers, payload }) => {
-          assert.strictEqual(headers.host, `127.0.0.1:${agent.port}`)
-          assertObjectContains(payload, [[{
-            name: 'prisma.client',
-            resource: 'User.create',
-            service: 'node-prisma',
-          }], [{
-            name: 'prisma.engine',
-            service: 'node-prisma',
-            meta: {
-              'db.user': 'postgres',
-              'db.name': 'postgres',
-              'db.type': 'postgres',
+        beforeEach(async function () {
+          this.timeout(60000)
+          agent = await new FakeAgent().start()
+          const commands = [
+            './node_modules/.bin/prisma migrate reset --force',
+            './node_modules/.bin/prisma db push --accept-data-loss',
+            './node_modules/.bin/prisma generate'
+          ]
+          const cwd = sandboxCwd()
+
+          if (config.ts) {
+            commands.push(
+              './node_modules/.bin/tsc ./generated/**/*.ts' +
+              ' --outDir ./dist' +
+              ' --target ES2023' +
+              ' --module ESNext' +
+              ' --strict true' +
+              ' --moduleResolution node' +
+              ' --esModuleInterop true'
+            )
+          }
+
+          execSync(commands.join(' && '), {
+            cwd,
+            stdio: 'inherit',
+            env: {
+              ...process.env,
+              ...config.env
+            }
+          })
+
+          // node v18 needs the package.json to have type module to treat .js files as esm
+          if (config.ts && config.name.includes('v6')) {
+            const fs = require('fs')
+            const path = require('path')
+            const distPath = path.join(cwd, 'dist')
+            if (!fs.existsSync(distPath)) {
+              fs.mkdirSync(distPath, { recursive: true })
             }
```

Comment:
```suggestion
            try {
              fs.mkdirSync(distPath, { recursive: true })
            } catch {}
```

That way there is no race condition and it should be faster